### ISSUE TITLE
Change Request: export an api to validate configs

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#17176]
### ESLint version

main

### What problem do you want to solve?

Currently, there is no straightforward method to validate ESLint configurations directly through ESLint itself. Developers have to rely on the runtime errors or manually checking the rules against the documentation to ensure their configurations are valid. By providing an API and a CLI option, we can streamline this process, reducing potential errors and improving efficiency for ESLint users.

### What do you think is the correct solution?
I am proposing the introduction of a new feature that could enhance the experience for developers working with ESLint configurations. The feature includes two main components:
1. API for Config Validation:
I suggest exposing an API that checks if a configuration is correct and verifies that all settings within the configuration are valid values. This API would be extremely helpful for developers and maintainers of shared configurations, as it would allow them to write tests for their configurations directly. Here is a potential use case illustrated with some example code:

```js
import { validateConfig } from "eslint";
import myConfig from "my-config";

test("export a valid config", () => {
    assert.doesNotThrow(() => validateConfig(myConfig));
});
```
2. CLI Option for Config Validation
Additionally, it would be beneficial to have a CLI option, tentatively named `--validate-config`, which allows users to check if their `eslint.config.js` is set up correctly. This would be particularly useful right after writing or modifying the ESLint configuration file. Here’s how it could be used:
```bash
eslint --validate-config
```

I am open to further suggestions on the naming and specifics of the implementation. I believe these additions would significantly improve the developer experience and aid in maintaining high-quality code standards.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
For reference, there was an accepted but never implemented RFC: https://github.com/eslint/rfcs/blob/main/designs/2019-config-tester/README.md. Though it seems that many things would not apply to the new flat config system, so I think it might be better to create a new RFC if there's a consensus on accepting this feature.

**Comment by mdjermanovic** [comment]
> ```js
> validateConfig(config); // throws if it's invalid
> ```

Can you give a brief overview of how this would work?

**Comment by nzakas** [comment]
I'm not opposed to the idea, but I would like some more details as to how this is expected to work. What sort of validation are we looking for, specifically?

**Comment by aladdin-add** [comment]
yes! I can see its value for config authors - just need a little time to see how to implement it. 😄

**Comment by nzakas** [comment]
It would also help if you could describe more fully what type of functionality you're looking for.

**Comment by aladdin-add** [comment]
@nzakas i've filled more info, sorry for the delaying!

**Comment by kecrily** [comment]
This seems to be essentially the same as https://github.com/eslint/rfcs/tree/main/designs/2019-config-tester? eslint-config-inspector also provides similar features. I think it looks good to standardize this and then eslint-config-inspector can use it.

**Comment by nzakas** [comment]
@aladdin-add there's still not a lot of information in your description. What specifically are you looking to validate? What problem is it that you're trying to avoid?
